LibreHardwareMonitorLib/Hardware/Controller/Corsair/CommanderPro.cs

```diff
@@ -0,0 +1,162 @@
+﻿using HidSharp;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LibreHardwareMonitor.Hardware.Controller.Corsair
+{
+    internal sealed class CommanderPro : Hardware
+    {
+        private List<Sensor> _fanSensors = new();
+        private List<Sensor> _tempSensors = new();
+        private List<Control> _controlSensors = new();
+        private ConcurrentQueue<ICommanderProCommand> _commands = new();
+        private List<ICommanderProCommand> _updateCommands = new();
+        private ushort[] _rpmBackup = new ushort[6];
+        private Task _task;
+        private StringBuilder _report = new();
+        private CommanderProDevice _device;
+
+        public CommanderPro(HidDevice dev, ISettings settings) : base("Commander Pro", new Identifier(dev.DevicePath), settings)
+        {
+            try
+            {
+                if (CommanderProDevice.TryOpen(dev, out CommanderProDevice device))
+                {
+                    _device = device;
+                    _report.AppendLine($"Commander Pro device product Id: {dev.ProductID}");
+
+                    foreach (int i in device.GetDetectedFanIndexes())
+                    {
+                        Sensor fan = new($"Commander Pro Fan #{i}", i, SensorType.Fan, this, settings);
+                        Sensor controlSensor = new("Commander Pro Fan Control #1", i, SensorType.Control, this, settings);
```

Comment:
why did you create another sensor for controlling? You can append the control to the fan.

Reply:
Because RPM and % are 2 different sensors on every other LHM implemented fan AFAIK.

Reply:
> Because the first byte is for the report number of an hid write/read call. 

Cool, added a span extension to only get the relevant part of the buffer and always start to read/write from 0.

Reply:
$"Commander Pro Fan Control #{i}"